src/register/pmpcfgx.rs

```diff
@@ -1,7 +1,129 @@
 /// Physical memory protection configuration
+
 pub mod pmpcfg0 {
-    read_csr_as_usize!(0x3A0, __read_pmpcfg0);
-    write_csr_as_usize!(0x3A0, __write_pmpcfg0);
+    use bit_field::BitField;
+
+    #[derive(Clone, Copy, Debug)]
+    pub enum Permission {
+        NONE = 0,
+        R = 1,
+        W = 2,
+        RW = 3,
+        X = 4,
+        RX = 5,
+        WX = 6,
+        RWX = 7,
+    }
+
+    #[derive(Clone, Copy, Debug)]
+    pub enum Range {
+        OFF = 0,
+        TOR = 1,
+        NA4 = 2,
+        NAPOT = 3,
+    }
+
+    #[derive(Clone, Copy, Debug)]
+    pub struct Pmpconfig {
+        pub permission: Permission,
+        pub range_type: Range,
+        pub locked: bool,
+    }
+
+    #[derive(Clone, Copy, Debug)]
+    pub struct Pmpcfg0 {
+        bits: usize,
+    }
+
+    impl Pmpcfg0 {
+        ///Returns the pmp byte associated with the index
+        #[inline]
+        pub fn pmp_byte(&self, index: usize) -> usize {
+            assert!(index < 8);
+            match index {
+                0 => self.bits.get_bits(0..8),
+                1 => self.bits.get_bits(8..16),
+                2 => self.bits.get_bits(16..24),
+                3 => self.bits.get_bits(24..32),
+                4 => self.bits.get_bits(32..40),
+                5 => self.bits.get_bits(40..48),
+                6 => self.bits.get_bits(48..56),
+                7 => self.bits.get_bits(56..64),
+                _ => panic!(),
```

Comment:
If rv64 arch, pmp4cfg to pmp7cfg will be in pmpcfg0 register, I'm not sure how to check this.

Reply:
`#[cfg(riscv32)]`, `#[cfg(riscv64)]`

Reply:
I added these to condition the assert! line, not sure that is correct.